### PR TITLE
Enable nullability in some ToolStrip members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some ToolStrip members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8429)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/StatusStrip.cs [hunk 1]
```diff
@@ -235,7 +235,7 @@ public new bool Stretch
 
         private TableLayoutSettings TableLayoutSettings
         {
-            get { return (TableLayoutSettings)LayoutSettings; }
+            get { return (TableLayoutSettings)LayoutSettings!; }
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/StatusStrip.cs:238]: I see the get for LayoutSettings is nullable, how do we know it's not null here?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/StatusStrip.cs:238]: This private property seems to be always called when we have `LayoutStyle == ToolStripLayoutStyle.Table`. So, I assumed that this is not null when used.

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 2]
```diff
@@ -1577,15 +1573,16 @@ public bool Stretch
...
         {
             get
             {
                 if (IsDropDown)
                 {
                     // PERF: since this is called a lot we don't want to make it virtual
-                    ToolStripDropDown dropDown = this as ToolStripDropDown;
-                    if (dropDown is ToolStripOverflow || dropDown.IsAutoGenerated)
+                    ToolStripDropDown? dropDown = this as ToolStripDropDown;
+                    if (dropDown is ToolStripOverflow || (dropDown is not null && dropDown.IsAutoGenerated))
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:1585]: It is a dropDown and to keep the behavior unchanged,  I would change it with direct cast.
```suggestion
                    var dropDown = (ToolStripDropDown)this;
                    if (dropDown is ToolStripOverflow || dropDown.IsAutoGenerated)
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 3]
```diff
@@ -1608,7 +1605,7 @@ public ToolStripRenderer Renderer
                     Renderer = ToolStripManager.CreateRenderer(RenderMode);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:1605]: We can replace this with.
```suggestion
                    _renderer = value;
                    _currentRendererType = _renderer.GetType();
                    OnRendererChanged(EventArgs.Empty);
```

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**